acme/acme/client.py

```diff
@@ -603,10 +603,8 @@ def query_registration(self, regr):
             Resource.
 
         """
-        self.net.account = regr
-        updated_regr = super(ClientV2, self).query_registration(regr)
-        self.net.account = updated_regr
-        return updated_regr
+        self.net.account = self._send_recv_regr(regr, None)
```

Comment:
Let's add a line of comment explaining why it's important to have it pass `None` instead of an empty `UpdateRegistration` message

Reply:
Added a comment here.

Reply:
Actually, I just noticed that there's a whole separate POST-as-GET function. Once we've removed the retry and deprecated ACMEv1 codepaths, we'll probably want this code to have been calling directly into the POST-as-GET function so we can keep those grouped together, no? Does the spec say that these calls should be done with POST-as-GET specifically, or does it say "post but with an empty body?

Reply:
Yes certainly we will want to do that. The spec says something more closer to the first assertion than the second one. It says that POST-as-GET is supposed to be used everywhere you are supposed to fetch a ressource, without saying it explicitly for the account. I suppose the migration path here is to make fetching accounts be done in a consistent way with the other resources.

Reply:
Is there a reason to not fetch accounts in a consistent way with other resources in this batch of changes?

Reply:
I do not understand your question. When I was talking about a consistent way in my previous response, I was refering to the fact that the specification in its current state turns out to require to fetch all ressources with a POST-as-GET request.

If your question is about why `_post_as_get()` function is not use here, you answered yourself previously: to be done in the future when there is no fallback to GET anymore an ACMEv1.

Reply:
Ah, I see there's some grammatical confusion. My earlier comments meant that once we've cleaned everything up, we will want to not need to make further changes; that is, to right now put the code in a state such that when we make that future change, everything will fit nicely together, and we won't have to go back and fix this. So to do this PR in a way that means we won't have to touch this codepath again in the future. Is that not possible?

Reply:
Ok I see. Indeed, with this in mind, the best design is to use directly a `_post()` then a `_regr_from_response()`. It is then consistent with the other calls where a `_post_as_get()` is done then a `_*_from_response()` (like `_authzr_from_response()`.

This way, when the `_post_as_get()` will stop to fallback to a `_get()`, we can simply replace the `_post()` by a `_post_as_get()` in `query_registration`.

I did that.

Reply:
That's a great idea, thank you!